catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -74,15 +87,22 @@ public RemoteSolrCatalogProvider(
       SolrClient client,
       SolrFilterDelegateFactory solrFilterDelegateFactory,
       @Nullable DynamicSchemaResolver resolver) {
-    this.provider =
-        new SolrCatalogProvider(
-            client,
-            filterAdapter,
-            solrFilterDelegateFactory,
-            (resolver == null) ? new DynamicSchemaResolver() : resolver);
+    this.filterAdapter = filterAdapter;
+    this.client = client;
+    this.solrFilterDelegateFactory = solrFilterDelegateFactory;
+    this.resolver = resolver;
+    init();
     provider.maskId(getId());
   }
 
+  protected void init() {
```

Comment:
❗️ This method is meant to finalize the initialization of the object via its constructor and as such, it must be private. Making it protected allows subclasses to override it and be called before their own constructor is ever called.